WordPress/openverse
# Centralise frontend error reporting (and suppress unactionable Sentry errors)

## Fixes



Fixes #3468 by @krysal

Superceeds #3721 after rebasing the branch to include #3811.

## Description


Introduces a new plugin to centralise network error handling. This allows us to make large-scale changes to how we handle certain classes of networking errors under certain conditions. To resolve the problems in the issue, this PR ignores the types of errors mentioned there, or sends information about them to a place we can better understand them (Plausible). For many ignored errors, we have other mechanisms to inspect and understand them, like API logs, or Cloudflare traffic analysis.

## Testing Instructions

Confirm the new tests sufficiently verify the behaviour of the plugin.

Run the app locally, and visit a search page. Open your browser dev tools network tab, and navigate to a single result. Find the "Related" request for the result, and block it using your dev tools (in Firefox you right click and "Block this URL" from the context menu). Navigate back to search, reload to clear the pinia cache of the related request, then go back to the same single result page. Confirm that the dev console has a log event for an analytics event, NETWORK_ERROR, and that no Sentry event was sent. You can confirm the sentry events by adding a console log at that line so you can see it in your nuxt dev console.

## Checklist


- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
